hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/ReplicationManager.java

```diff
@@ -543,11 +543,15 @@ private void handleUnderReplicatedContainer(final ContainerInfo container,
         List<DatanodeDetails> targetReplicas = new ArrayList<>(source);
         // Then add any pending additions
         targetReplicas.addAll(replicationInFlight);
-
-        int delta = replicationFactor - getReplicaCount(id, replicas);
         final ContainerPlacementStatus placementStatus =
             containerPlacement.validateContainerPlacement(
                 targetReplicas, replicationFactor);
+        int delta = replicationFactor - getReplicaCount(id, replicas);
+        if (placementStatus.isPolicySatisfied() && delta <= 0) {
```

Comment:
Rather than this new IF block here, would it make sense to simply add:

```
if (replicasNeeded <= 0) {
  LOG.debug(...);
  return;
}

At line 554 / 558, just after the line:

```
        final int replicasNeeded
            = delta < misRepDelta ? misRepDelta : delta;
```

That would avoid needing to check call `placementStatus.isPolicySatisfied()` and then `placementStatus.misReplicationCount()`.

Reply:
Make sense.